components/backdrop/backdrop.js

```diff
@@ -0,0 +1,184 @@
+import '../colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { getComposedChildren, getComposedParent } from '../../helpers/dom.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+
+const scrollKeys = [];
+let scrollOverflow = null;
+
+class Backdrop extends LitElement {
+
+	static get properties() {
+		return {
+			shown: { type: Boolean },
+			forTarget: { type: String, attribute: 'for-target' },
+			_state: { type: String, reflect: true }
+		};
+	}
+
+	static get styles() {
+		return [ css`
+			:host {
+				background-color: var(--d2l-color-regolith);
+				height: 0;
+				left: 0;
+				opacity: 0;
+				position: fixed;
+				top: 0;
+				transition: opacity 200ms ease-in;
+				width: 0;
+				z-index: 999;
+			}
+			:host([_state="showing"]), :host([_state="hiding"]) {
+				height: 100%;
+				width: 100%;
+			}
+			:host([_state="showing"]) {
+				opacity: 0.7;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.shown = false;
+		this._state = null;
+	}
+
+	attributeChangedCallback(name, oldval, newval) {
+		super.attributeChangedCallback(name, oldval, newval);
+
+		if (name !== 'shown') return;
+
+		if (this.shown) this._show();
+		else this._hide(true);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this.shown) {
+			this._hide(false);
+		}
+	}
+
+	render() {
+		return html`<div></div>`;
+	}
+
+	_hide(animate) {
+
+		const hide = () => {
+			if (animate) this.removeEventListener('transitionend', hide);
+
+			allowBodyScroll(this._bodyScrollKey);
+			this._bodyScrollKey = null;
+
+			showAccessible(this._hiddenElements);
+			this._hiddenElements = null;
+			this._state = null;
+		};
+
+		if (animate) {
+			this.addEventListener('transitionend', hide);
+			this._state = 'hiding';
+		} else {
+			hide();
+		}
+
+	}
+
+	_show() {
+
+		this._bodyScrollKey = preventBodyScroll();
+		this._hiddenElements = hideAccessible(this.parentNode.querySelector(`#${this.forTarget}`));
+		this._state = 'showing';
+
+	}
+
+}
+
+export function allowBodyScroll(key) {
```

Comment:
I refactored the logic for preventing body scroll so that our native dialog logic can make use of it, since native dialog backdrop does not prevent scrolling on the body.  The `preventBodyScroll` method may get called more than once - for instance, with nested dialogs.